test/controllers/api/search_controller_test.exs

```diff
@@ -22,8 +22,27 @@ defmodule Constable.Api.SearchesControllerTest do
     assert results_for(conn, query: query_with_extra_spaces) == json_for(announcement_2)
   end
 
+  test "excludes announcements that have an excluded interest", %{conn: conn} do
+    lame = insert(:interest, name: "lame")
+    announcement_1 = insert(:announcement, title: "foobar1") |> tag_with_interest(lame)
+    announcement_2 = insert(:announcement, body: "announcement body cool")
+    announcement_3 = insert(:announcement, title: "awesome title", body: "cool body")
+
+    assert results_for_exclude(conn, query: "foobar1", exclude: ["lame"]) == json_for([])
+    assert results_for_exclude(conn, query: "announcement body", exclude: ["lame"]) == json_for(announcement_2)
+    assert results_for_exclude(conn, query: "awesome title cool body", exclude: []) == json_for(announcement_3)
+  end
+
   defp results_for(conn, query: search_term) do
-    response = post conn, api_search_path(conn, :create), query: search_term
+    results_for_exclude(conn, query: search_term, exclude: [])
+  end
+
+  defp results_for_exclude(conn, query: search_term, exclude: exclude) do
```

Comment:
And maybe it could use the same function name as above? `results_for(conn, query: "announcement_body", exclude_interests: ["lame"])`?